test/AtomicTokenSwap.js

```diff
@@ -0,0 +1,142 @@
+
+      await instance.initiate(now + day, hashLock, other, token.address, 100);
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it(`should fail if swap doesn't exist`, async () => {
+    try {
+      const instance = await atomicTokenSwap.deployed();
+      await instance.redeem("some-inexistent-secret");
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it (`should fail if the message sender is not the swap participant`, async () => {
+    try {
+      const instance = await atomicTokenSwap.deployed();
+      await instance.redeem(secret);
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it('should let you redeem your tokens from the swap', async () => {
+    const instance = await atomicTokenSwap.deployed();
+    const token = await testToken.deployed();
+
+    await instance.redeem(secret, {from: other});
+    const balance = await token.balanceOf(other);
+
+    const {5: exists} = await instance.swaps(other, hashLock);
+
+    assert.equal(balance, 100);
+    assert.equal(exists, false);
+  });
+
+  it(`should not refund if swap doesn't exist for participant`, async () => {
+    try {
+      const instance = await atomicTokenSwap.deployed();
+      await instance.refund(hashLock, owner);
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it(`should not refund if swap doesn't exist for participant with that hash`, async () => {
+    try {
+      const instance = await atomicTokenSwap.deployed();
+      await instance.refund("wrong-hahs-lock", other);
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it(`should not refund if exipration time was not met`, async () => {
+    try {
+      const instance = await atomicTokenSwap.deployed();
+      await instance.refund(hashLock, other);
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it(`should not refund if refundee is not the initiator`, async () => {
+    try {
+      const instance = await atomicTokenSwap.deployed();
+
+      await timeTravel(day);
+
+      await instance.refund(hashLock, other, {from: notInitiator});
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it(`should refund if all requirements are met`, async () => {
+    const instance = await atomicTokenSwap.deployed();
+    const token = await testToken.deployed();
+
+    const beforeBalance = await token.balanceOf(owner);
+
+    await timeTravel(day);
+    await timeTravel(day);
+
+    await instance.refund(hashLock, other);
```

Comment:
Throws a revert even tho I don't understand, tested manually on remix and it worked as expected.

Reply:
Oops, nevermind.. Haha. Let me think this over...